osu.Game/Overlays/BeatmapSet/MetadataSectionNominators.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Resources.Localisation.Web;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public partial class MetadataSectionNominators : MetadataSection<(BeatmapSetOnlineNominations[] CurrentNominations, APIUser[] RelatedUsers)>
+    {
+        public override (BeatmapSetOnlineNominations[] CurrentNominations, APIUser[] RelatedUsers) Metadata
+        {
+            set
+            {
+                if (value.CurrentNominations.Length == 0)
+                {
+                    this.FadeOut(TRANSITION_DURATION);
+                    return;
+                }
+
+                base.Metadata = value;
+            }
+        }
+
+        public MetadataSectionNominators(Action<(BeatmapSetOnlineNominations[] CurrentNominations, APIUser[] RelatedUsers)>? searchAction = null)
+            : base(MetadataType.Nominators, searchAction)
+        {
+        }
+
+        protected override void AddMetadata((BeatmapSetOnlineNominations[] CurrentNominations, APIUser[] RelatedUsers) metadata, LinkFlowContainer loaded)
+        {
+            int[] nominatorIds = metadata.CurrentNominations.Select(n => n.UserId).ToArray();
+
+            int nominatorsFound = 0;
+
+            foreach (int nominatorId in nominatorIds)
+            {
+                foreach (var user in metadata.RelatedUsers)
+                {
+                    if (nominatorId != user.OnlineID) continue;
+
+                    nominatorsFound++;
+
+                    loaded.AddUserLink(new APIUser
+                    {
+                        Username = user.Username,
+                        Id = nominatorId,
+                    });
+
+                    if (nominatorsFound < nominatorIds.Length)
+                        loaded.AddText(CommonStrings.ArrayAndWordsConnector);
+
+                    break;
+                }
+            }
+        }
```

Comment:
I may be alone on this but I would prefer encapsulating this logic in `APIBeatmapSet` itself, rather than throwing everything at the UI-side component (with an awkward tuple) and letting it figure out which users that are actually nominators (@bdach interested in your thought).

Reply:
There is precedent to the approach you're proposing ([`CommentBundle`](https://github.com/ppy/osu/blob/master/osu.Game/Online/API/Requests/Responses/CommentBundle.cs)). That said I'm not necessarily very keen on that because if you start jamming that logic in there, (a) the structure becomes even more bloated than before, (b) you suddenly have to start caring about things like deserialisation order in order to keep the logic working properly.

I say leave as it is as a local implementation until it becomes a problem somewhere else. Chances are, it will not.